apps/mobile/src/libs/components/radio-group/radio-group.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { useState } from "react";
+import { RadioGroup as RNRadioGroup } from "react-native-radio-buttons-group";
+
+import { Text } from "~/libs/components/components";
+import { BaseColor } from "~/libs/enums/enums";
+import { globalStyles } from "~/libs/styles/styles";
+import { type RadioButtonProps, type ViewStyle } from "~/libs/types/types";
+
+import { DefaultStyles } from "./libs/constants/constants";
+import { styles } from "./styles";
+
+type RadioGroupItemStyle = {
+	circleBorderColorActive?: string;
+	circleBorderSize?: number;
+	circleBorderSizeActive?: number;
+	circleColorActive?: string;
+	circleSize?: number;
+} & ViewStyle;
+
+type RadioGroupItem = {
+	label: string;
+	value: string;
+};
+
+type Properties = {
+	itemContainerStyle?: RadioGroupItemStyle;
+	onChange?: (value: RadioGroupItem) => void;
+	options: RadioGroupItem[];
+};
+
+const RadioGroup: React.FC<Properties> = ({
+	itemContainerStyle = {},
+	onChange,
+	options,
+}: Properties) => {
+	const {
+		circleBorderColorActive = BaseColor.BG_BLUE,
+		circleBorderSize = DefaultStyles.DEFAULT_BORDER_SIZE,
+		circleBorderSizeActive = DefaultStyles.DEFAULT_ACTIVE_BORDER_SIZE,
+		circleColorActive = BaseColor.BG_WHITE,
+		circleSize = DefaultStyles.DEFAULT_CIRCLE_SIZE,
+	} = itemContainerStyle;
+
+	const [selectedId, setSelectedId] = useState<string>("");
+
+	const radioButtonsProperties: RadioButtonProps[] = [];
+	const itemsMap = new Map<string, RadioGroupItem>();
```

Comment:
Since every answer will have its `id` we don't need this map.

Reply:
The Map is needed to store the association radio_button_id and input data because the button ID is not the same as the input data ID.

By generating a synthetic button ID, I encapsulated the radio button's behavior. I excluded the issue when the user (the person who uses the component) makes a mistake when passing the wrong ID in the data.